### ISSUE TITLE
Possible bug in moon mag calculation

### ISSUE THREAD

**Issue by alexander-lapshin** [Bill-Gray/lunar#8]
Hi Bill!

Looks like there is an error in file "vislimit.cpp" line 166 
`
b->lunar_mag = -12.73 + moon_elong * (.026 +
                           4.e-9 * (moon_elong * moon_elong * moon_elong));
`
there lunar_mag is brightest (-12.73)  when moon_elong equals zero, but zero moon_elong is the new moon.

Best regards,
Alexander Lapshin

**Comment by Bill-Gray** [comment]
Hi Alexander,

Hmmm... thank you,  you're right;  that should be something along the lines of 'moon_dist_from_opposition',  a value that is 180 degrees minus the "real" moon elongation.  I'm surprised that didn't turn up before.

Fortunately,  the fix is fairly simple.  The elongation is the value passed in,  and 'moon_angle' is the value we really want and use later on to compute `b->lunar_mag`.  (That appears to be the only use that is made of the lunar elongation.)

This function has been used in two places.  One is in a desktop planetarium program I wrote years ago;  I used it to generate a 'realistic' night sky.  I think I did the 180-degree swap there and therefore had a brighter night sky at full moon.

The other place it's used is in my Find_Orb orbit determination software... and there I see that I mangled things a bit,  in such a way as to get (usually) fairly realistic-looking values,  but they were really brighter than they ought to be for points close to the moon.  It shouldn't be hard to fix.

I'll mention that there is a [(slightly) better lunar magnitude vs. elongation function](https://www.projectpluto.com/groups/guide-user/msg3410.htm) (really an empirical table;  I don't know of any more recent version),  and we could adjust the lunar magnitude for distance from the moon and sun (simple inverse-square).  I don't think it would really improve the sky brightness/visibility calculation by much,  but it might give a slight improvement to the computed lunar magnitude,  which might be useful elsewhere.
